src/main/java/org/alfresco/metrics/rest/RestMetricsReporterImpl.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * #%L
+ * Alfresco Remote API
+ * %%
+ * Copyright (C) 2005 - 2016 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software. 
+ * If the software was purchased under a paid Alfresco license, the terms of 
+ * the paid license agreement will prevail.  Otherwise, the software is 
+ * provided under the following open source license terms:
+ * 
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.metrics.rest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.alfresco.micrometer.MetricsController;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+
+public class RestMetricsReporterImpl implements RestMetricsReporter, ApplicationContextAware, InitializingBean
+{
+    private Log logger = LogFactory.getLog(getClass());
+    public static final String METER_REGISTRY_BEAN_ERROR_MESSAGE = "Could not get the meterRegistry bean, essential for reporting REST metrics.";
+    private static final String REST_EXECUTION_TIME = "rest-api.execution.time";
+    private static final int MAX_TAG_LENGTH = 1024;
+
+    private boolean restMetricsEnabled;
+    private boolean restServicePathMetricsEnabled;
+
+    private ApplicationContext applicationContext;
```

Comment:
You should not need this. 
Just get a reference to metricsController from spring
see: 
https://github.com/Alfresco/alfresco-repository/blob/master/src/main/java/org/alfresco/repo/metrics/db/DBMetricsReporterImpl.java#L62